### ISSUE TITLE
(contributing): The content of integration testing document isn't stable after remodeling

### ISSUE THREAD

**Issue by yo-ga** [aws/aws-cdk#24958]
### Describe the issue

After the related remodeling task, the integration testing document isn't updated.
It makes the developers confused and hard to follow.

### Links

[INTEGRATION_TESTS.md](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)

**Comment by peterwoodworth** [comment]
I've been having some issues with our documentation in general since the change, lots of broken links everywhere! i'm going to use this issue to consolidate all broken links if that's okay with you 🙂

**Comment by yo-ga** [comment]
@peterwoodworth  Any will do.

**Comment by plumdog** [comment]
Main thing I've hit is the links from the main docs that link to the source code, eg:
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html

Which has a "(source)" link next to "TypeScript". This links to https://github.com/aws/aws-cdk/blob/v2.73.0/packages/@aws-cdk/aws-s3/lib/bucket.ts#L1623, the old location in the repo. (This location existed as of 2.72.0 but is now gone).

**Comment by yo-ga** [comment]
Actually, this original scope is more likely the same as the meaning for the label `doc/repo-general`.
@peterwoodworth  If you are like to collect and track the whole documented issues, here the better way is.
You create the tracking issue on your own. After doing that, you link the issues, including this one.

**Comment by peterwoodworth** [comment]
Also this issue #26037

**Comment by peterwoodworth** [comment]
https://github.com/aws/aws-cdk/issues/26662
